Pcap++/src/PcapFileDevice.cpp

```diff
@@ -114,6 +118,93 @@ int IFileReaderDevice::getNextPackets(RawPacketVector& packetVec, int numOfPacke
 	return numOfPacketsRead;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// SnoopFileReaderDevice members
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+SnoopFileReaderDevice::~SnoopFileReaderDevice()
+{
+	m_snoopFile.close();
+}
+
+bool SnoopFileReaderDevice::open()
+{
+	m_NumOfPacketsRead = 0;
+	m_NumOfPacketsNotParsed = 0;
+
+	m_snoopFile.open(m_FileName.c_str(), std::ifstream::binary);
+	snoop_file_header_t snoop_file_header;
+	m_snoopFile.read((char*)&snoop_file_header, sizeof(snoop_file_header_t));
+
+	if(be64toh(snoop_file_header.identification_pattern) != 0x736e6f6f70000000 && be32toh(snoop_file_header.version_number) == 2)
+		return false;
+
+	// From https://datatracker.ietf.org/doc/html/rfc1761
+	static const pcpp::LinkLayerType snoop_encap[] = {
+		LINKTYPE_ETHERNET,	/* IEEE 802.3 */
+		LINKTYPE_NULL,		/* IEEE 802.4 Token Bus */
+		LINKTYPE_IEEE802_5,	/* IEEE 802.5 */
+		LINKTYPE_NULL,		/* IEEE 802.6 Metro Net */
+		LINKTYPE_ETHERNET,	/* Ethernet */
+		LINKTYPE_C_HDLC,	/* HDLC */
+		LINKTYPE_NULL,		/* Character Synchronous, e.g. bisync */
+		LINKTYPE_NULL,		/* IBM Channel-to-Channel */
+		LINKTYPE_FDDI		/* FDDI */
+	};
+
+	m_PcapLinkLayerType = snoop_encap[be32toh(snoop_file_header.datalink_type)];
+
+	PCPP_LOG_DEBUG("Successfully opened file reader device for filename '" << m_FileName << "'");
+	m_DeviceOpened = true;
+	return true;
+}
+
+void SnoopFileReaderDevice::getStatistics(PcapStats& stats) const
+{
+	stats.packetsRecv = m_NumOfPacketsRead;
+	stats.packetsDrop = m_NumOfPacketsNotParsed;
+	stats.packetsDropByInterface = 0;
+	PCPP_LOG_DEBUG("Statistics received for reader device for filename '" << m_FileName << "'");
+}
+
+bool SnoopFileReaderDevice::getNextPacket(RawPacket& rawPacket)
+{
+	rawPacket.clear();
+	if (m_DeviceOpened != true)
+	{
+		PCPP_LOG_ERROR("File device '" << m_FileName << "' not opened");
+		return false;
+	}
+        snoop_packet_header_t snoop_packet_header;
+        m_snoopFile.read((char*)&snoop_packet_header, sizeof(snoop_packet_header_t));
+        if(!m_snoopFile) {
+            return false;
+        }
+        size_t packetSize = be32toh(snoop_packet_header.included_length);
+        if(packetSize > 15000) {

```

Comment:
just curious: why is the 15000-byte limitation (and not some other number)?
I don't think we should change it, just curious why you chose 15000...

Reply:
To be honest, I just added something big - I would assume the largest possible packet would be 9000 Jumbo frame